cdci_data_analysis/analysis/instrument.py

```diff
@@ -299,9 +299,23 @@ def run_query(self, product_type,
                     logger.warning("bad request from user, passing through: %s", e)
                     raise
                 except Exception as e: # we shall not do that
+                    message_prepend_str = ('Unfortunately, the analysis product you requested is empty, '
```

Comment:
this should be returned for a subset of exceptions, deriving from `UnfortunateRequestResults`: this means request lead to something apparently unfortunate. It's not bad, just empty, as it should be. Nothing wrong with that! Let's not shame users if their requests lead to nothing! It should be advised in the message that the user may want to consider another rrequest, jus tas written below.

In addition, we should also catch `BadRequest`. This means something is inconsistent, not interprettable by us in the request: e.g. parameter does not comply with format, out of range, etc.

Finally, other exceptions should be like `InternalError`. This should raise sentry issue and advise user that we are looking into it, it is worth retrying , and it might get better.